lib/stores/localstore.ts

```diff
@@ -0,0 +1,391 @@
+import {
+  DurablePromise,
+  PendingPromise,
+  ResolvedPromise,
+  RejectedPromise,
+  CanceledPromise,
+  TimedoutPromise,
+  isPendingPromise,
+  isResolvedPromise,
+  isRejectedPromise,
+  isCanceledPromise,
+} from "../promise";
+import { IPromiseStore } from "../store";
+import { IEncoder } from "../encoder";
+import { Base64Encoder } from "../encoders/base64";
+import { ErrorCodes, ResonateError } from "../error";
+
+export class LocalPromiseStore implements IPromiseStore {
+  private readonly dbName = "resonateDB";
+  private readonly storeName = "promises";
+  private db: IDBDatabase | null = null;
```

Comment:
It would be nice if this could be just type `IDBDatabase` and whoever creates an instance of this class immediately gets an error if there is a problem getting/creating the db.

What do you think about this?
1. pass the IDBDatabase instance to the constructor
2. this pushes the problem to the user, so to mitigate this we could add a static method to this class (which can be async) where we first instantiated the `IDBDatabase` and then return an instance of the local promise store?

Reply:
Great advice.